applications/vanilla/models/class.commentmodel.php

```diff
@@ -1207,10 +1235,52 @@ public function save($formPostValues, $settings = false) {
                 $this->updateCommentCount($formPostValues['DiscussionID'], ['Slave' => false]);
             }
         }
-
+        $comment = $commentID ? $this->getID($commentID, DATASET_TYPE_ARRAY): false;
+        if ($comment) {
+            $commentEvent = $this->eventFromRow(
+                (array)$comment,
```

Comment:
We're already specifying this row should be represented as an array, a few lines up. We shouldn't need to cast it here.

```suggestion
                $comment,
```